main.tf

```diff
@@ -5,11 +5,37 @@ provider "aws" {
 resource "aws_s3_bucket" "demo-s3" {
   bucket = "${var.s3_bucket_prefix}-terraformcloud"
   versioning {
-    enabled = false
-    mfa_delete = false
```

Comment:
**Accurics remediation**
 
**[HIGH]: S3 bucket should have versioning MFA delete enabled**
 
**Vulnerability**
Enabling MFA delete for versioning is a good way to add extra protection to sensitive files stored in buckets.aws s3api put-bucket-versioning --bucket bucketname --versioning-configuration Status=Enabled,MFADelete=Enabled --mfa your-mfa-serial-number mfa-code
 
**Remediation**
Using the AWS s3api CLI, enable MFA Delete for the S3 buckets that fail this rule, for example: aws s3api put-bucket-versioning --bucket bucketname --versioning-configuration Status=Enabled,MFADelete=Enabled --mfa your-mfa-serial-number mfa-code. Refer to: https://aws.amazon.com/blogs/security/securing-access-to-aws-using-mfa-part-3

source: accurics-